File: goimpl/parser/return.go
package parser

import (
	"goimpl/ast"
	"goimpl/token"
)

var parseReturnStatment parseStatement = func(p *Parser) (ast.Statement, error) {
	stmt := &ast.Return{Tok: p.currTok}
	//FIXME: Implement the rest of the parsing logic, skipping expression at the moment
	for p.nextToken(); p.currTok.Type != token.SEMICOLON; p.nextToken() {
	}
	return stmt, nil
}

File: scala-impl/src/main/scala/com/monkeylang/parser/Return.scala
package com.monkeylang.parser

import com.monkeylang.lexer.Lexer
import com.monkeylang.ast.Statement.*
import com.monkeylang.ast.Expression.*

protected val returnParser: Lexer -> (Return, Lexer) = (l: Lexer) =>
  (Return(Some(BooleanLiteral("true"))), l)
